### PR TITLE
5693 geolocation window

### PR DESCRIPTION
**Pull request by kellykelly3**

### CODE REVIEW

**Comment by raman-bt** [comment]
Codacy didn't run on this PR?

File: Core/src/org/sleuthkit/autopsy/geolocation/GeolocationTopComponent.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_TRACKPOINT));
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_SEARCH));
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_LAST_KNOWN_LOCATION));
+                artifacts.addAll(currentCase.getSleuthkitCase().getBlackboardArtifacts(BlackboardArtifact.ARTIFACT_TYPE.TSK_GPS_BOOKMARK));
+                
+                for (BlackboardArtifact artifact : artifacts) {
+                    Waypoint point = new SimpleArtifactWaypoint(artifact);
+
+                    if (point.getPosition() != null) {
+                        waypoints.add(new SimpleArtifactWaypoint(artifact));
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/GeolocationTopComponent.java]: we cant do waypoints.add(point) here. 

no need to create same artifact again?

File: Core/src/org/sleuthkit/autopsy/geolocation/GeolocationUtilities.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/GeolocationUtilities.java]: Update file header.

File: Core/src/org/sleuthkit/autopsy/geolocation/GeolocationUtilities.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+     * @param type     The attribute type we're looking for
+     *
+     * @return The Double if it exists, or null if not
+     */
+    static Double getDouble(BlackboardArtifact artifact, BlackboardAttribute.ATTRIBUTE_TYPE type) {
+        Double returnValue = null;
+        try {
+            BlackboardAttribute bba = artifact.getAttribute(new BlackboardAttribute.Type(type));
+            if (bba != null) {
+                Double value = bba.getValueDouble();
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/GeolocationUtilities.java]: Any specific reason for this two step assignment?  
Couldn't returnValue  be directly assigned to?

(applies to other utilities methods in the class)

File: Core/src/org/sleuthkit/autopsy/geolocation/RefreshPanel.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/RefreshPanel.java]: Update file header.

File: Core/src/org/sleuthkit/autopsy/geolocation/RefreshPanel.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+package org.sleuthkit.autopsy.geolocation;
+
+import java.awt.event.ActionListener;
+
+/**
+ *
+ * 
+ */
+public class RefreshPanel extends javax.swing.JPanel {
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/RefreshPanel.java]: Update  class header above.

File: Core/src/org/sleuthkit/autopsy/geolocation/Route.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+package org.sleuthkit.autopsy.geolocation;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.jxmapviewer.viewer.GeoPosition;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+
+/**
+ * A route is a set of Geolocations with a common parent artifact;
```
Comment (raman-bt) [inline Core/src/org/sleuthkit/autopsy/geolocation/Route.java]: Editorial comment: remove/replace the ';' 

(applies to other methods in this file)

**Comment by rcordovano** [comment]
@raman-bt and @kellykelly3, Codacy has run on this PR now. There are issues. Also a Travis build failure.

File: Core/src/org/sleuthkit/autopsy/geolocation/GeolocationUtilities.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                returnValue = value;
+            }
+        } catch (TskCoreException ex) {
```
[the remark this answers is not in the document]
Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/geolocation/GeolocationUtilities.java]: @raman-bt and @kellykelly3, Raman is correct. Lower level classes need to leave error handling policy to the top level classes. I would expect a GeolocationUtilities class to have no logging.

[the remark this answers is not in the document]
Comment (kellykelly3) [reply Core/src/org/sleuthkit/autopsy/geolocation/GeolocationUtilities.java]: I'll just delete the whole class.  I copied and pasted the functions from somewhere else and what I liked about them was that they handled the exceptions.

**Pull request closed without merging**